src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs

```diff
@@ -23,107 +22,321 @@ internal static class TrustedSignersCommand
-                    {
-                        throw new CommandLineArgumentCombinationException(string.Format(CultureInfo.CurrentCulture, Strings.Error_UnknownAction, command.Values[0]));
-                    }
+        private static async Task<int> ExecuteCommand(TrustCommand action,
+                      CommandOption algorithm,
+                      bool allowUntrustedRootOption,
+                      CommandOption owners,
+                      CommandOption verbosity,
+                      CommandOption configFile,
+                      Func<ILogger> getLogger,
+                      Action<LogLevel> setLogLevel,
+                      string name = null,
+                      string sourceUrl = null,
+                      string packagePath = null,
+                      string fingerprint = null)
+        {
+            ILogger logger = getLogger();
 
-                    string name = null;
+            try
+            {
+                ISettings settings = ProcessConfigFile(configFile.Value());
 
-                    if (command.Values.Count > 1)
-                    {
-                        name = command.Values[1];
-                    }
+                var trustedSignersArgs = new TrustedSignersArgs()
+                {
+                    Action = MapTrustEnumAction(action),
+                    PackagePath = packagePath,
+                    Name = name,
+                    ServiceIndex = sourceUrl,
+                    CertificateFingerprint = fingerprint,
+                    FingerprintAlgorithm = algorithm?.Value(),
+                    AllowUntrustedRoot = allowUntrustedRootOption,
+                    Author = action == TrustCommand.Author,
+                    Repository = action == TrustCommand.Repository,
+                    Owners = CommandLineUtility.SplitAndJoinAcrossMultipleValues(owners?.Values),
+                    Logger = logger
+                };
+
+                setLogLevel(XPlatUtility.MSBuildVerbosityToNuGetLogLevel(verbosity.Value()));
+
+#pragma warning disable CS0618 // Type or member is obsolete
+                var sourceProvider = new PackageSourceProvider(settings, enablePackageSourcesChangedEvent: false);
+#pragma warning restore CS0618 // Type or member is obsolete
+                var trustedSignersProvider = new TrustedSignersProvider(settings);
 
-                    string packagePath = null;
-                    string sourceUrl = null;
-                    string fingerprint = null;
-                    if (command.Values.Count() > 2)
+                var runner = new TrustedSignersCommandRunner(trustedSignersProvider, sourceProvider);
+                Task<int> trustedSignTask = runner.ExecuteCommandAsync(trustedSignersArgs);
+                return await trustedSignTask;
+            }
+            catch (Exception e)
+            {
+                // nuget trust command handled exceptions.
+                if (e is InvalidOperationException)
```

Comment:
I think we should handle `InvalidOperationException` and `ArgumentException` in seperate `catch` blocks instead of handling all the exceptions in the generic exception handler. 

How about making subcommands responsible for throwing appropriate exceptions and here we just `log the error` and return `exit code: 1`?

Reply:
I separated the try catch blocks.
Not sure subcommands throwing appropriate exception, it's just rethrowing exception, more work but end result is almost same.